### PR TITLE
Lay out structure for property/method renames for v6.0

### PR DESCRIPTION
**Pull request by kattrali**
## Goal

This changeset lays out an initial structure for the updates necessary to get the library ready for v6.0.0. The primary change in this major revision will be to rename some properties and methods to better align with other Bugsnag notifier libraries. These changes have been added in a backwards-compatible way to allow for incremental releases leading up to v6, when all deprecated properties will be removed. 

Every change should be listed in the changelog and upgrade migration guide, and the new properties should replace the existing ones in the integration guide on docs.bugsnag.com.

### CODE REVIEW

File: Source/BugsnagSessionTracker.m [hunk 1]
```diff
@@ -91,7 +91,7 @@ - (BugsnagSession *)runningSession {
 }
 
 - (void)startNewSessionIfAutoCaptureEnabled {
-    if (self.config.shouldAutoCaptureSessions  && [self.config shouldSendReports]) {
+    if (self.config.autoDetectSessions && [self.config shouldSendReports]) {
```
Comment (fractalwrench) [inline Source/BugsnagSessionTracker.m]: I can't find the exact section in the spec but believe one of the points of discussion is that options like `shouldSendReports` should no longer affect sessions. This assumes that `shouldSendReports` has similar semantics to `autoDetectErrors` - is this the case? We may just want to remove the option entirely.

File: Source/Bugsnag.h [hunk 2]
```diff
@@ -200,7 +200,8 @@ static NSString *_Nonnull const BugsnagSeverityInfo = @"info";
  *
  * @param capacity max number of breadcrumb log messages to send
  */
-+ (void)setBreadcrumbCapacity:(NSUInteger)capacity;
++ (void)setMaxBreadcrumbs:(NSUInteger)capacity;
```
Comment (fractalwrench) [inline Source/Bugsnag.h]: This is not on the Bugsnag interface any more according to the spec, and should be on Configuration instead.

File: Source/Bugsnag.m [hunk 3]
```diff
@@ -215,12 +215,16 @@ + (void)leaveBreadcrumbForNotificationName:
     }
 }
 
-+ (void)setBreadcrumbCapacity:(NSUInteger)capacity {
++ (void)setMaxBreadcrumbs:(NSUInteger)capacity {
     if ([self bugsnagStarted]) {
         self.notifier.configuration.breadcrumbs.capacity = capacity;
```
Comment (fractalwrench) [inline Source/Bugsnag.m]: The default capacity is now 25 according to the spec, with bounds of 0-100 - this should be enforced.

File: UPGRADING.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Upgrading
+
+Guide to ease migrations between significant changes
+
+## v5 -> v6
+
+Version 6 introduces a number of property and method renames:
```
Comment (tomlongridge) [inline UPGRADING.md]: Does it matter if this is included in master before the v6 release?

Comment (kattrali) [reply UPGRADING.md]: Not particularly, though we could cut some beta releases.

**Pull request merged**